### PR TITLE
[test] Fix Firefox browser version in karma profile config and resolve user-event test TODOs

### PR DESCRIPTION
**Pull request by ZeeshanTamboli**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/base-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Typo spotted from https://github.com/mui/base-ui/pull/179 while working on https://github.com/mui/material-ui/pull/42008.

Not certain if we've ever executed [performance monitoring tests](https://github.com/mui/material-ui/blob/next/test/README.md#performance-monitoring).

Additionally, I considered refactoring these TODOs:

- https://github.com/mui/material-ui/blob/next/packages/mui-base/src/MenuButton/MenuButton.test.tsx#L10-L12
- https://github.com/mui/material-ui/blob/next/packages/mui-base/src/Select/Select.test.tsx#L21-L23

now that we support the new Safari version, but I believe these components will undergo a refactor with a new API, so I left them as is.

### CODE REVIEW

**Comment by michaldudak** [comment]
Thanks!
As for the tests, we'd like to reuse as much as we can, so refactoring them will be useful anyway.

**Comment by ZeeshanTamboli** [comment]
> Thanks! As for the tests, we'd like to reuse as much as we can, so refactoring them will be useful anyway.

Okay, I resolved the TODOs. Here's the `browserstack-force` run - https://app.circleci.com/pipelines/github/mui/base-ui/1317/workflows/61c67cfd-9b6f-408c-9fe0-60c6f99fad69/jobs/7582.

File: packages/mui-base/src/MenuButton/MenuButton.test.tsx [hunk 1]
```diff
@@ -155,7 +151,9 @@ describe('<MenuButton />', () => {
             button.focus();
           });
 
-          await userEvent.keyboard(`{${key}}`);
+          const user = userEvent.setup();
```
Comment (michaldudak) [inline packages/mui-base/src/MenuButton/MenuButton.test.tsx:154]: Do you know if this is required in each test and shouldn't be called just once at the top?

Comment (ZeeshanTamboli) [reply packages/mui-base/src/MenuButton/MenuButton.test.tsx]: I initially thought about this but I believed tests should be independent. However, upon your inquiry, I researched further and discovered relevant documentation: https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent. Additionally, I found a helpful explanation on Stack Overflow: https://stackoverflow.com/a/76457218/8146534. 

Moreover, I came across this insightful discussion: https://github.com/testing-library/user-event/discussions/1052#discussioncomment-3641692. 

Furthermore, I followed the recommendation to invoke `setup()` before rendering the component.

Apologies for the numerous commits; I encountered some issues with pnpm v8.

New `browserstack-force` CI run for Safari: https://app.circleci.com/pipelines/github/mui/base-ui/1345/workflows/d42c306c-32ef-4237-af04-90d77a371586/jobs/7691

Comment (michaldudak) [reply packages/mui-base/src/MenuButton/MenuButton.test.tsx]: Fair enough, thank you!

**Review by michaldudak — APPROVED** [review]

**Pull request merged**